pmlc/util/layout.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 Intel Corporation
+#include "pmlc/util/layout.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+
+#include "mlir/IR/BuiltinTypes.h"
+
+namespace pmlc {
+
+using namespace llvm; // NOLINT
+using namespace mlir; // NOLINT
+
+MLFramework getMLFramework(StringRef opName) {
+  StringRef nGraphPrefix = "ng.";
+  if (opName.substr(0, nGraphPrefix.size()) == nGraphPrefix) {
+    return MLFramework::NGraph;
+  } else {
+    return MLFramework::Default;
+  }
+}
+
+TensorLayout getLayoutType(MLFramework framework, StringRef opName,
+                           bool isConst) {
+  // TODO: Create a common list of primitives that would be used here and in the
+  // plugins so we know they are consistent. For now use namings from OV plugin
+  // Const are used to identify the primitives weights or parameters, rest is
+  // considered as main flow data type
+  if (framework == MLFramework::NGraph) {
```

Comment:
Good idea. Generally this is done in a very basic way now, there are few things to consider like the common base of the primitives (right now the layer names are defined in the plaidml bridge, some start with capital letters, some do not, and do not have consistent namings). It would be good to auto generate these somehow and keep consistency with the plugins. I can come up with some proposition for this, but if you have something specific in mind then let me know